score/confirm.go

```diff
@@ -33,5 +40,32 @@ func CalculateConfirmScore(metric *schema.Metric) {
 	if denominator > 0 {
 		score = 1 - numerator/denominator
 	}
+
+	return score
+}
+
+func CalculateConfirmScore(metric *schema.Metric) {
+	details := &metric.Details.Confirm
+	dataset := details.ContinuousData
+
+	sizeOfConfirmData := len(dataset)
+	if 0 == sizeOfConfirmData {
+		metric.Details.Confirm.Score = 0
+		metric.Details.Confirm.ScoreYesterday = 0
+		return
+	} else if sizeOfConfirmData < consts.ConfirmScoreWindowSize {
+		dataset = datasetPrependZero(dataset)
+		details.ContinuousData = dataset
+	}
+
+	datasetYesterday := details.ContinuousData[0 : len(details.ContinuousData)-1]
```

Comment:
`details.ContinuousData` comes from here: https://github.com/bitmark-inc/autonomy-api/blob/4526ecbc9d2587294402e2eeb54952081832089c/store/metric.go#L91.

`consts.ConfirmScoreWindowSize` is 14. The first data point for calculating the score is not retrieved even if exists.

Reply:
I have checked that function already. In the function, there is a line
```
opts = options.Find().SetSort(bson.M{"report_ts": -1}).SetLimit(windowSize + 1)
```

I do not know the reason why it uses `windowSize + 1` but it is like there will be enough points for me.

Reply:
I can't recall why it's `windowSize + 1` now. @pieceofr Do you have any clue?

This PR now looks to me. I might want to add comments for `ContinuousDataCDSConfirm` to highlight the length of returned slice is actually `windowSize + 1` instead of `windowSize`, though this is beyond the scope of current PR.

Reply:
@nodestory @pieceofr If so, that means the calculation is totally wrong at this moment.

Reply:
I think I will not address that problem here.